### PR TITLE
feat: Set variables on Apple M1 to build Python packages

### PR DESCRIPTION
**Pull request by armenzg**
This sets certain flags on the developer's machine in order to allow building confluent-kafka and uWSGI.

Setting the flags globally under `~/.zshrc` benefits other project that depend on these Python libraries (e.g. Snuba needs confluent-kafka since sentry-arroyo is used in the pre-commit hooks).

### CODE REVIEW

File: scripts/lib.sh [hunk 1]
```diff
@@ -105,31 +105,52 @@ install-py-dev() {
...
-        pip install https://storage.googleapis.com/python-arm64-wheels/confluent_kafka-1.5.0-cp38-cp38-macosx_11_0_arm64.whl
-        # uwsgi does not properly install via pyenv Python installations
-        # https://github.com/unbit/uwsgi/issues/2361
-        LDFLAGS=-L/opt/homebrew/Cellar/gettext/0.21/lib  pip install "uWSGI==2.0.19.1"
     fi
+
+    # SENTRY_LIGHT_BUILD=1 disables webpacking during setup.py.
+    # Webpacked assets are only necessary for devserver (which does it lazily anyways)
+    # and acceptance tests, which webpack automatically if run.
+    SENTRY_LIGHT_BUILD=1 pip install -e '.[dev]'
```
Comment (armenzg) [inline scripts/lib.sh:121]: I'm only moving the `fx_profile` block after this. It only makes sense to patch *after* the package has been installed. See #29887 for original work.

File: scripts/lib.sh [hunk 2]
```diff
@@ -105,31 +105,52 @@ install-py-dev() {
...
+    # The CPATH is needed for confluent-kakfa --> https://github.com/confluentinc/confluent-kafka-python/issues/1190
+    # The LDFLAGS is needed for uWSGI --> https://github.com/unbit/uwsgi/issues/2361
+    body="
+$header
+export CPATH=/opt/homebrew/Cellar/librdkafka/1.8.2/include:\${CPATH}
+export LDFLAGS=-L/opt/homebrew/Cellar/gettext/0.21/lib:\${LDFLAGS}"
+    if [ "$SHELL" == "/bin/zsh" ]; then
+        if ! grep -qF "${header}" "${zshrc_path}"; then
+            echo "Added the following to ${zshrc_path}"
+            cp "${zshrc_path}" "${zshrc_path}.bak"
```
Comment (armenzg) [inline scripts/lib.sh:146]: We make a copy just in case. I blew my own when I was doing development for this.

File: scripts/lib.sh [hunk 3]
```diff
@@ -211,6 +232,7 @@ build-platform-assets() {
 }
 
 bootstrap() {
+    setup-apple-m1
```
Comment (armenzg) [inline scripts/lib.sh:235]: This is important for all engineers to execute once.

File: scripts/lib.sh [hunk 4]
```diff
@@ -105,31 +105,52 @@ install-py-dev() {
...
+export LDFLAGS=-L/opt/homebrew/Cellar/gettext/0.21/lib:\${LDFLAGS}"
+    if [ "$SHELL" == "/bin/zsh" ]; then
+        if ! grep -qF "${header}" "${zshrc_path}"; then
+            echo "Added the following to ${zshrc_path}"
+            cp "${zshrc_path}" "${zshrc_path}.bak"
+            echo -e "$body" >> "${zshrc_path}"
+            echo -e "$body"
+        fi
+    else
+        echo "You are not using a supported shell. Please add these variables where appropiate."
```
Comment (armenzg) [inline scripts/lib.sh:151]: For anyone not using `zsh` I will leave it up to them rather than guess their preference.

File: scripts/lib.sh [hunk 5]
```diff
@@ -106,30 +106,51 @@ install-py-dev() {
     # It places us within top src dir to be at the same path as setup.py
     # This helps when getsentry calls into this script
     cd "${HERE}/.." || exit
+
     echo "--> Installing Sentry (for development)"
     if query-apple-m1; then
         # This installs pyscopg-binary2 since there's no arm64 wheel
         # This saves having to install postgresql on the Developer's machine + using flags
         # https://github.com/psycopg/psycopg2/issues/1286
         pip install https://storage.googleapis.com/python-arm64-wheels/psycopg2_binary-2.8.6-cp38-cp38-macosx_11_0_arm64.whl
```
Comment (armenzg) [inline scripts/lib.sh:115]: This package was built differently than the other two and we won't be able to work around it. We could get away with it if we installed the postgresql library on the host via brew.

**Review by markstory — APPROVED** [review]
I was able to get sentry and snuba dependencies to build with these changes.

File: scripts/lib.sh [hunk 6]
```diff
@@ -106,30 +106,51 @@ install-py-dev() {
...
+    ! query-apple-m1 && return
+
+    zshrc_path="${HOME}/.zshrc"
+    header="# Apple M1 environment variables"
+    # The CPATH is needed for confluent-kakfa --> https://github.com/confluentinc/confluent-kafka-python/issues/1190
+    # The LDFLAGS is needed for uWSGI --> https://github.com/unbit/uwsgi/issues/2361
+    body="
+$header
+export CPATH=/opt/homebrew/Cellar/librdkafka/1.8.2/include
+export LDFLAGS=-L/opt/homebrew/Cellar/gettext/0.21/lib"
```
[the remark this answers is not in the document]
Comment (billyvg) [reply scripts/lib.sh:142]: fwiw it does get checked here https://github.com/getsentry/sentry/pull/30094/files#diff-5838719667e50a69f27dfddac131cad8ed4e11bee597a64a1b48454efdfad40bR50-R52

**Pull request merged**